### ISSUE TITLE
Quadrotors sharing paylod

### ISSUE THREAD

**Issue by PedroHRPBS** [ethz-asl/rotors_simulator#387]
Hello, guys

I have a project  to develop where I  have 2 quads connected with a bar.

https://ibb.co/efGc3v

In the image above you can see the project illustrated using 1 quad as an example.
I'm having a lot of trouble trying to implement this using Hector_quadrotor, so I tought about switching to this simulator.
The main issues that I had was to make the spherical joint between the  quad and the bar, and also having control of  the two aircrafts at the same time. First because it uses URDF which doesn't implement spherical joints and also because having control of two quads in the same model requires some Plugin editing that I don't know how to do.

Anyone has any idea if I can solve these problems with this sim?

Thanks in advance!

**Comment by ffurrer** [comment]
@fmina has worked on similar problems.
So just pinging him here, he might be able to give you some hints.

**Comment by fmina** [comment]
To create a spherical joint you will have to combine three revolute joints. 
Controlling two MAVs doesn't need any modification of the plugins. You just have to use correct namespace for each MAV, for example `firefly1` and `firefly2`.

**Comment by PedroHRPBS** [comment]
But then, how do I add the bar to connect them? Changing the namespace is done on the .launch file, right? Is it possible to create joints there?

**Comment by fmina** [comment]
No, two MAVs connected with one object is seen by gazebo as single robot. You need to prepare a urdf file that creates the mechanics of two MAVs and connect them with joints and links. The MAV mechanics is defined here: https://github.com/ethz-asl/rotors_simulator/blob/master/rotors_description/urdf/firefly.xacro

**Comment by PedroHRPBS** [comment]
So, to have a "robot" that constains 2 quads, I would have to instantiate 2 quads inside of this file that you sent? I want that the two quads fly independently.

**Comment by juanmed** [comment]
@PedroHRPBS  Hi. I see this thread has been around for a long time. Did you by any chance get to work this around? Joining two quads with a bar? If yes, would you mind pointing to any examples on how you got it working? Thanks!

**Comment by PedroHRPBS** [comment]
@juanmed actually I gave up, this was a project that I was working on at my graduation. But it took another path, so I didn't end up solving this. Sorry.

**Comment by juanmed** [comment]
I see. Thanks for your answer!
